assets/src/components/libraries/SearchBar.test.jsx

```diff
@@ -10,13 +11,14 @@ describe('SearchBar', () => {
   const onChange = jest.fn();
 
   it('should render with empty search term', () => {
-    searchBarComponent = shallow(<SearchBar onChange={onChange} query="" />);
-    expect(searchBarComponent.state().searchTerm).toEqual('');
+    searchBarComponent = render(<SearchBar onChange={onChange} query="" />);
+    const input = searchBarComponent.getByPlaceholderText('Search by book title or author');
+    expect(input.value).toBe('');
   });
 
   it('should render with passed search term', () => {
-    searchBarComponent = shallow(<SearchBar onChange={onChange} query="test search" />);
-    expect(searchBarComponent.state().searchTerm).toEqual('test search');
+    searchBarComponent = render(<SearchBar onChange={onChange} query="test search" />);
+    searchBarComponent.getByDisplayValue('test search');
   });
 
   it('should clear search term when click in close icon', () => {
```

Comment:
I could't reproduce the same test using RTL. I'm having problems while clicking on the <Close/> component. I used the `screen.logTestingPlaygroundURL() ` in order to check how the element can be retrieve so we can click on it, but it didn't work as well :| 

Do you have any idea on how to solve this?

Reply:
Yeah, it's because the button right now is only being rendered as an image instead of a button 😞. You can see the rendered tree rendering the component and calling `searchBarComponent.debug()`.

You can fix this if you set the icon role to `button` and use the `titleAccess` prop to [describe its meaning semantically](https://v3.material-ui.com/style/icons/#semantic-svg-icons), like this (which is also great for accessibility):

Replace `<Close className="search-bar-close-icon" onClick={this.onClear} />`
with `<Close className="search-bar-close-icon" onClick={this.onClear} role="button" titleAccess="Clear" />`

And then you can test it by using:

```
    const clearButton = searchBarComponent.getByRole('button', { title: /clear/i });
    fireEvent.click(clearButton);

    const input = searchBarComponent.getByPlaceholderText(/search/i);
    expect(input.value).toBe('');
```

We could also replace the usage of the icon directly to wrap it in a `IconButton` like we do in the header, but you'd need to adjust some css bits.

Reply:
Awesome! Thanks for your help, I'll try it on. You talked about the `.debug()` thing. I've heard recently about a new feature for testing library where you can check your component being rendered on testing playground so you can identify easily how to get a component from screen. I thing it's pretty helpful for us. Check it out if you want to: https://twitter.com/kentcdodds/status/1316044932327469056?s=09 

Off course we must update the testing library version so we can take advantage on that.

Reply:
Omg!!! This is so awesome! Thanks for the tip @erikaya :)